valuable-derive/src/expand.rs

```diff
@@ -135,13 +147,22 @@ fn derive_enum(input: &syn::DeriveInput, data: &syn::DataEnum) -> TokenStream {
                     .iter()
                     .map(|field| field.ident.as_ref())
                     .collect();
+                let as_value = variant.fields.iter().map(|field| {
+                    let f = field.ident.as_ref();
+                    let tokens = quote! {
+                        // HACK(taiki-e): This `&` is not actually needed to calling as_value,
+                        // but is needed to emulate multi-token span on stable Rust.
+                        &#f
```

Comment:
On stable Rust, if remove this `&`, error message only points to the first token.
```text
error[E0277]: the trait bound `S: Valuable` is not satisfied
  --> $DIR/not_valuable.rs:15:17
   |
15 |     Struct { f: Option<S> },
   |                 ^^^^^^ the trait `Valuable` is not implemented for `S`
   |
```